[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-complex-if

enforce split big conditional statement to variables

## Rule details

✔️ Examples of **correct** code:
```js
if (a && b) {}
if (a && b || c) {}
if (a && (b || c)) {}
if (a && (b || c && d)) {}
```

❌ Examples of **incorrect** code with `[{"depth":4}]` options:
```js
if (a && (b && c && (d || c))) {}
if ((a && c || (d && n)) && b || c) {}
if (a || c && b || d && c || d) {}
```

## Resources

* [Rule source](/lib/rules/no-complex-if.js)
* [Test source](/tests/lib/rules/no-complex-if.js)
