[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-complex-if

No complex if

üîß The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## Rule details

‚úîÔ∏è Examples of **correct** code:
```js
if (a && b) {}
if (a && b || c) {}
if (a && (b || c)) {}
if (a && (b || c && d)) {}
```

‚ùå Examples of **incorrect** code with `[{"depth":4}]` options:
```js
if (a && (b && c && (d || c))) {}
if ((a && c || (d && n)) && b || c) {}
if (a || c && b || d && c || d) {}
```

## Resources

* [Rule source](/lib/rules/no-complex-if.js)
* [Test source](/tests/lib/rules/no-complex-if.js)
