[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-ternary-assigment

No assigment in ternary expression

## Rule details

❌ Examples of **incorrect** code:
```js
a === 1 ? b = 1 : null;
true ? c() : c = 0
```

✔️ Examples of **correct** code:
```js
c = 0;
a === 1 ? b : c;
true ? c() : null;
if (a) { c = 3 }
```

## Resources

* [Rule source](/lib/rules/no-ternary-assigment.js)
* [Test source](/tests/lib/rules/no-ternary-assigment.js)
