[//]: # (This file is generated by eslint-docgen. Do not edit it directly.)

# no-ternary-assigment

disable assigment in ternary expression

## Rule details

❌ Examples of **incorrect** code:
```js
a === 1 ? b = 1 : null;
true ? c() : c = 0;
true ? c ? a = b : g : h;
addressData === false ? addressData = {} : logger = false;
```

✔️ Examples of **correct** code:
```js
a === 1 ? b : c;
true ? c() : null;
var c = a ? 1 : 2
```

## Resources

* [Rule source](/lib/rules/no-ternary-assigment.js)
* [Test source](/tests/lib/rules/no-ternary-assigment.js)
